Userland/Libraries/LibCore/EventLoop.h

```diff
@@ -84,6 +87,8 @@ class EventLoop {
     void post_event(Object& receiver, NonnullOwnPtr<Event>&&, ShouldWake = ShouldWake::No);
     void wake_once(Object& receiver, int custom_event_type);
 
+    void add_job(NonnullRefPtr<Promise<NonnullRefPtr<Object>>> job_promise);
```

Comment:
Why does this have to be a `Promise<NNRP<Object>>`? That seems like an arbitrary limitation.

Reply:
As discussed with Ben (see above), the other solution is some sort of type erasure and/or void*.

Reply:
Type erasure sounds like a better solution than BackgroundAction::Result and inheritance from Promise

Reply:
Considering my experience with type erased data pointers in LibGUI models, I will very strongly disagree.

Reply:
EventLoop doesn't care about the exact type of the Promise, the caller does. We already have the map() function on promise to handle changing a PromiseA into a PromiseB

Reply:
Plus, we already have type erasure in play, of the inheritance kind. see <https://github.com/SerenityOS/serenity/pull/16715#discussion_r1117999962>